src/cephadm/cephadmlib/container_types.py

```diff
@@ -687,24 +667,46 @@ def get_mgr_images() -> dict:
     """Return dict of default mgr images"""
     mgr_prefix = 'mgr/cephadm/container_image_'
     mgr_images = {}
-    mgr_images[mgr_prefix + 'prometheus'] = DEFAULT_PROMETHEUS_IMAGE
-    mgr_images[mgr_prefix + 'alertmanager'] = DEFAULT_ALERT_MANAGER_IMAGE
-    mgr_images[mgr_prefix + 'graphana'] = DEFAULT_GRAFANA_IMAGE
-    mgr_images[mgr_prefix + 'loki'] = DEFAULT_LOKI_IMAGE
-    mgr_images[mgr_prefix + 'promtail'] = DEFAULT_PROMTAIL_IMAGE
-    mgr_images[mgr_prefix + 'node_exporter'] = DEFAULT_NODE_EXPORTER_IMAGE
-    mgr_images[mgr_prefix + 'haproxy'] = DEFAULT_HAPROXY_IMAGE
-    mgr_images[mgr_prefix + 'keepalived'] = DEFAULT_KEEPALIVED_IMAGE
-    mgr_images[mgr_prefix + 'nvmeof'] = DEFAULT_NVMEOF_IMAGE
-    mgr_images[mgr_prefix + 'snmp_gateway'] = DEFAULT_SNMP_GATEWAY_IMAGE
-    mgr_images[mgr_prefix + 'elasticsearch'] = DEFAULT_ELASTICSEARCH_IMAGE
+    mgr_images[
+        mgr_prefix + 'prometheus'
+    ] = default_images.DEFAULT_PROMETHEUS_IMAGE
+    mgr_images[
+        mgr_prefix + 'alertmanager'
+    ] = default_images.DEFAULT_ALERT_MANAGER_IMAGE
+    mgr_images[mgr_prefix + 'graphana'] = default_images.DEFAULT_GRAFANA_IMAGE
+    mgr_images[mgr_prefix + 'loki'] = default_images.DEFAULT_LOKI_IMAGE
+    mgr_images[
+        mgr_prefix + 'promtail'
+    ] = default_images.DEFAULT_PROMTAIL_IMAGE
+    mgr_images[
+        mgr_prefix + 'node_exporter'
+    ] = default_images.DEFAULT_NODE_EXPORTER_IMAGE
+    mgr_images[mgr_prefix + 'haproxy'] = default_images.DEFAULT_HAPROXY_IMAGE
+    mgr_images[
+        mgr_prefix + 'keepalived'
+    ] = default_images.DEFAULT_KEEPALIVED_IMAGE
+    mgr_images[mgr_prefix + 'nvmeof'] = default_images.DEFAULT_NVMEOF_IMAGE
+    mgr_images[
+        mgr_prefix + 'snmp_gateway'
+    ] = default_images.DEFAULT_SNMP_GATEWAY_IMAGE
+    mgr_images[
+        mgr_prefix + 'elasticsearch'
+    ] = default_images.DEFAULT_ELASTICSEARCH_IMAGE
```

Comment:
This could simplified/rewritten as: 

```
image_suffixes = [
    'prometheus',
    'alertmanager',
    'graphana',
    'loki',
    'promtail',
    'node_exporter',
    'haproxy',
    'keepalived',
    'nvmeof',
    'snmp_gateway',
    'elasticsearch',
]

for suffix in image_suffixes:
    key = mgr_prefix + suffix
    value = getattr(default_images, f'DEFAULT_{suffix.upper()}_IMAGE')
    mgr_images[key] = value

```

Reply:
swapped to your second suggestion and tested it and it works nicely. It also helped discover a bug in the previous code as the config option name for grafana was incorrect (spelled "graphana") and the some of the SMB names differed as well.

```
[root@vm-00 ~]# diff old.txt new.txt 
3,4c3,4
< mgr/cephadm/container_image_alertmanager = quay.io/prometheus/alertmanager:v0.27.0
< mgr/cephadm/container_image_graphana = quay.io/ceph/grafana:10.4.8
---
> mgr/cephadm/container_image_node_exporter = quay.io/prometheus/node-exporter:v1.7.0
> mgr/cephadm/container_image_nvmeof = quay.io/ceph/nvmeof:1.2.17
7c7,8
< mgr/cephadm/container_image_node_exporter = quay.io/prometheus/node-exporter:v1.7.0
---
> mgr/cephadm/container_image_alert_manager = quay.io/prometheus/alertmanager:v0.27.0
> mgr/cephadm/container_image_grafana = quay.io/ceph/grafana:10.4.8
10d10
< mgr/cephadm/container_image_nvmeof = quay.io/ceph/nvmeof:1.2.17
15,17d14
< mgr/cephadm/container_image_jaeger_query = quay.io/jaegertracing/jaeger-query:1.29
< mgr/cephadm/container_image_smb = quay.io/samba.org/samba-server:devbuilds-centos-amd64
< mgr/cephadm/container_image_smbmetrics = quay.io/samba.org/samba-metrics:latest
19a17,19
> mgr/cephadm/container_image_jaeger_query = quay.io/jaegertracing/jaeger-query:1.29
> mgr/cephadm/container_image_samba = quay.io/samba.org/samba-server:devbuilds-centos-amd64
> mgr/cephadm/container_image_samba_metrics = quay.io/samba.org/samba-metrics:latest
```

Reply:
I did have to modify the alertmanager variable name for this to work as it had a "_" between alert and manager before, but that's fine

Reply:
We're planning some further work to have the config setting for the images be specified in this same images.py file in the future which I think will cover that.